internal/controllers/topology/machinedeployment/machinedeployment_controller.go

```diff
@@ -55,14 +57,26 @@ type Reconciler struct {
 }
 
 func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
-	err := ctrl.NewControllerManagedBy(mgr).
-		For(&clusterv1.MachineDeployment{}).
+	clusterToMachineDeployments, err := util.ClusterToTypedObjectsMapper(mgr.GetClient(), &clusterv1.MachineDeploymentList{}, mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
+	err = ctrl.NewControllerManagedBy(mgr).
+		For(&clusterv1.MachineDeployment{},
+			builder.WithPredicates(
+				predicates.ResourceIsTopologyOwned(ctrl.LoggerFrom(ctx)),
+				predicates.ResourceNotPaused(ctrl.LoggerFrom(ctx)))).
 		Named("topology/machinedeployment").
+		WithEventFilter(predicates.ResourceHasFilterLabel(ctrl.LoggerFrom(ctx), r.WatchFilterValue)).
```

Comment:
I separated the combined filter `ResourceNotPausedAndHasFilterLabel`,  since IMO it is easier to read and we do not have any other Watched resources in this controller